### PR TITLE
LL-5272 Allow access to swap feature without the device wall

### PR DESCRIPTION
**Pull request by juan-cortes**
When accessing the swap feature we currently require users to have the exchange app installed and up to date and when inside the form we limit the selectors depending on whether they have the applications installed or not. This means we are capping the feature before users can access it. This feature allows the user to enter the swap form and request a form without knowing that they need the exchange app and, once the inline install is no longer experimental, they will be able to fulfill that requirement without leaving the form.

### Type
Feature

### Context
https://ledgerhq.atlassian.net/browse/LL-5272

### Parts of the app affected / Test plan
- Without experimental inline install, try cases with/without all required apps/accounts
- With experimental inline install, try cases with/without all required apps/accounts

### CODE REVIEW

File: src/renderer/screens/exchange/swap/Swap.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // @flow
 
-import React, { useMemo, useEffect, useCallback, useState } from "react";
+// import React, { useMemo, useEffect, useCallback, useState } from "react";
```
Comment (JunichiSugiura) [inline src/renderer/screens/exchange/swap/Swap.js:3]: ✂️

File: src/renderer/screens/exchange/swap/Form/ToAccount.js [hunk 2]
```diff
@@ -86,21 +83,12 @@ const ToAccount = ({
     }
   }, [account, accounts, currency, setAccount]);
 
-  const renderOptionOverride = useCallback(
-    ({ data: currency }: any) => {
-      // NB ignore the custom rendering for no-accounts here since we show the add account CTA
-      const status = currenciesStatus[currency.id];
+  const renderOptionOverride = useCallback(({ data: currency }: any) => {
+    // NB ignore the custom rendering for no-accounts here since we show the add account CTA
```
Comment (JunichiSugiura) [inline src/renderer/screens/exchange/swap/Form/ToAccount.js:87]: 👍

File: static/i18n/en/app.json [hunk 3]
```diff
@@ -1032,7 +1035,10 @@
     "unlockDevice": "Unlock your device",
     "unlockDeviceAfterFirmwareUpdate": "Wait for the firmware to update and unlock your device with your PIN",
     "quitApp": "Quit the application on your device",
-    "appNotInstalled": "Please install the {{appName}} app",
+    "appNotInstalledTitle": "Missing required app",
+    "appNotInstalledTitle_plural": "Missing required apps",
+    "appNotInstalled": "The {{appName}} app is required to complete this action. Please go to the manager and install it on your device",
+    "appNotInstalled_plural": "The {{appName}} apps are required to complete this action. Please go to the manager and install them on your device",
```
Comment (JunichiSugiura) [inline static/i18n/en/app.json:1041]: 👍

File: src/renderer/screens/exchange/swap/Form/FromAccount.js [hunk 4]
```diff
@@ -111,20 +121,28 @@ const FromAccount = ({
           value={currency}
           autoFocus={true}
           onChange={setCurrency}
-          isCurrencyDisabled={isCurrencyDisabled}
         />
       </Box>
       <Box>
         <Label mb={4} mt={25}>
-          <Trans i18nKey={`swap.form.from.account`} />
+          <Trans i18nKey={`swap.form.to.account`} />
```
Comment (JunichiSugiura) [inline src/renderer/screens/exchange/swap/Form/FromAccount.js:128]: Does it have to be dynamic? (from <-> to). I haven't check the UI so I'm not sure but.

**Review by JunichiSugiura — CHANGES_REQUESTED** [review]

File: src/renderer/screens/exchange/swap/Form/FromAccount.js (hunk 4 above)
Comment (juan-cortes) [reply src/renderer/screens/exchange/swap/Form/FromAccount.js:128]: It incidentally is the same wording, simply "Account" but you're right it should be "to"

File: src/renderer/screens/exchange/swap/Swap.js (hunk 1 above)
Comment (juan-cortes) [reply src/renderer/screens/exchange/swap/Swap.js:3]: oh yeah, nice catch

**Comment by nabil-brn** [comment]
When I access the swap feature, it quits the app on my nano.
I though we removed all device action which is not the case.
Did we forget something to remove it?

**Comment by juan-cortes** [comment]
> When I access the swap feature, it quits the app on my nano.
> I though we removed all device action which is not the case.
> Did we forget something to remove it?

Should not happen for sure, I'll check today

**Comment by juan-cortes** [comment]
As tested with @nabil-brn the bug is not really present and must have been a bad build somewhere.
Updated removing the import and changing the wording as per the review by @JunichiSugiura

**Review by JunichiSugiura — DISMISSED** [review]
LGTM assuming that CI tests will pass.
Can't wait to see it on the app. It'll greatly improve our UX.

**Review by nabil-brn — CHANGES_REQUESTED** [review]
I'm having an error when I reach inline app installation (without experimental enabled).

Prerequisites:
- On Portfolio: ETH and BTC accounts
- On Nano: ETH installed but NOT BTC

1. Swap
2. Perform a swap ETH -> BTC
3. During the flow it should tell you that you don't have BTC app installed
4. At this time check the logs

```
[17904:0519/143015.149:ERROR:CONSOLE(0)] "Error in event handler: Error: Cannot remove node "24666" because no matching node was found in the S
tore.
    at chrome-extension://bdmanjkbngoajfnopfikjpbnmmdeeapp/build/main.js:21477:43
    at bridge_Bridge.emit (chrome-extension://bdmanjkbngoajfnopfikjpbnmmdeeapp/build/main.js:19585:22)
    at chrome-extension://bdmanjkbngoajfnopfikjpbnmmdeeapp/build/main.js:19745:12
    at listener (chrome-extension://bdmanjkbngoajfnopfikjpbnmmdeeapp/build/main.js:37733:39)", source: chrome-extension://bdmanjkbngoajfnopfikj
```
---
a5cd1f086 (HEAD -> LL-5272, juan/LL-5272) Cleanup

**Review by Arnaud97234 — DISMISSED** [review]
**Experimental inline install ON**
- When app missing on device -> Apps are installed then user can confirm the swap
- When apps already on the device -> User can confirm the swap

**Experimental inline install OFF**
- When app is missing on the device -> User is redirected to the Manager with the search input filled with requested currencies
- When apps already on the device -> User can confirm the swap

@Nabil if you don't have any issue anymore we can merge ✅

**Review by valpinkman — DISMISSED** [review]

**Comment by nabil-brn** [comment]
This issue https://github.com/LedgerHQ/ledger-live-desktop/pull/3824#pullrequestreview-663139832 seems to be present in develop so, it's not related to this PR imo.

**Review by nabil-brn — DISMISSED** [review]

**Comment by nabil-brn** [comment]
👍 
---
a5cd1f086 (HEAD -> LL-5272, juan/LL-5272) Cleanup

**Review by Arnaud97234 — APPROVED** [review]

**Review by LFBarreto — APPROVED** [review]

**Review by JunichiSugiura — APPROVED** [review]

**Pull request merged**